### PR TITLE
fix(mission_planner, route_handler): enable loop detection of route

### PR DESCRIPTION
**Pull request by tkimura4**
Signed-off-by: tomoya.kimura <tomoya.kimura@tier4.jp>

## Description
In the current implementation, isRouteLooped() always returns false because one `primitives` has only one `id`.
Duplicate check of lane_id must be done for the entire `segments`. I fixed it.

## Tests performed
I confirmed that the following error occurs and route was not published when I tried to set route with overlapping lanes using check points.
image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
@mitsudome-r -san is added because I think he did it first.

File: planning/route_handler/src/route_handler.cpp [hunk 1]
```diff
@@ -199,9 +183,30 @@ void RouteHandler::setMap(const HADMapBin & map_msg)
   setLaneletsFromRouteMsg();
 }
 
+bool RouteHandler::isRouteLooped(const RouteSections & route_sections) const
```
Comment (TakaHoribe) [inline planning/route_handler/src/route_handler.cpp:186]: Is this implementation much simpler?

```cpp
bool isRouteLooped(const RouteSections & route_sections) const
{
  std::set<lanelet::Id> lane_primitives;
  for (const auto & route_section : route_sections) {
    for (const auto & primitive : route_section.primitives) {
      if (lane_primitives.find(primitive.id) == lane_primitives.end()) {
        lane_primitives.set(primitive.id);
      } else {
        return true;  // find duplicated id
      }
    }
  }
  return false;
}
```

Comment (tkimura4) [reply planning/route_handler/src/route_handler.cpp:186]: Thank you! It's good. I fixed.

**Review by TakaHoribe — APPROVED** [review]
LGTM!

**Comment by taikitanaka3** [comment]
@tkimura4 
can you resolve this conflict?

**Review by mitsudome-r — APPROVED** [review]

**Comment by tkimura4** [comment]
@taikitanaka3 
resolved!

**Review by taikitanaka3 — APPROVED** [review]
LGTM
I confirmed there is no warning like before this PR 
image

**Pull request merged**